concordium-consensus/src/Concordium/GlobalState/PoolRewards.hs

```diff
@@ -21,7 +21,9 @@ data BakerPoolRewardDetails = BakerPoolRewardDetails
       -- | The total transaction fees accrued to this pool in the reward period
       transactionFeesAccrued :: !Amount,
       -- | Whether the pool contributed to a finalization proof in the reward period
-      finalizationAwake :: !Bool
+      finalizationAwake :: !Bool,
+      -- | The number of missed rounds in the reward period
+      missedRounds :: !Word16
```

Comment:
I changed it to Word64. I think even Word16 is unlikely to overflow though. General question: Do we optimize on the size of serialized types?

Reply:
I also made the field optional. Another general question: What are the benefits of this current approach of checking for backwards compatibility in the type system, over just supporting one protocol version at HEAD and have well tested migration paths from the previous protocol version? I feel the effort is not proportional to the benefits at the moment, but maybe I'm missing something.

Reply:
Word16 shouldn't overflow with the current chain parameters, but we shouldn't rely on that.

I think it is generally good to optimize for space somewhat, but balance the amount of space saved vs complexity. In this case, I think it could be worth doing some space saving. For instance, `finalizationAwake` takes a whole byte of serialization, but only the low-order bit is used - you could use another bit to record if `missedRounds` is non-zero. Usually it will be zero, and as there's going to be at least one of these written per block, the savings could be worth while.

On the backwards compatibility question, I'm not entirely sure what the alternative you are suggesting is. In practise, the node would need to support at least 2 protocol versions at once in order to migrate. As it stands, the node can catch up from the original genesis to the latest block. If we were implementing a new node fresh today, maybe we wouldn't bother with the older protocol versions at all, and we might do a bunch of things differently from the outset. However, the node as it exists is a result of a series of incremental steps, and the approach to protocol versions that we have allows us to make them reasonably easily, in my opinion.

Reply:
> On the backwards compatibility question, I'm not entirely sure what the alternative you are suggesting is. In practise, the node would need to support at least 2 protocol versions at once in order to migrate. As it stands, the node can catch up from the original genesis to the latest block. If we were implementing a new node fresh today, maybe we wouldn't bother with the older protocol versions at all, and we might do a bunch of things differently from the outset. However, the node as it exists is a result of a series of incremental steps, and the approach to protocol versions that we have allows us to make them reasonably easily, in my opinion.

The alternative I'd suggest is more or less as you say: Track protocol versions only for persisted data, not for code. And support only the migration of persisted data from the last released version to the current version, while the implemented protocol is always just the newest one.

I believe this would lower the development time of new protocol features quiet a bit, lower code complexity and also compile times. I'd estimate that around 80% of code changes for the suspended validators feature is just concerned about keeping track of the new protocol version and turning it on only for P8. That's a lot of code we could avoid with the above approach. Also, how will this scale for a lot more protocol versions (imagine P20)? No need to discuss further here, as this is more a consideration if we'd ever start a fresh node (or for the light node).